train/train.py

```diff
@@ -284,13 +308,50 @@ def train():
 
     random.shuffle(raw_data)
 
-    dataset = CustomDataset(raw_data, tokenizer)
+    if training_args.do_eval:
+        # Take 90:10 train-validation split
+        raw_train_data, raw_eval_data = split_data(raw_data, data_args.data_path, 0.9)
+        train_dataset = CustomDataset(raw_train_data, tokenizer)
+        eval_dataset = CustomDataset(raw_eval_data, tokenizer)
+    else:
+        train_dataset = CustomDataset(raw_data, tokenizer)
+        
+    def preprocess_logits_for_metrics(logits, labels):
```

Comment:
Can you type concise docstrin on what this code does? Also please refactor this function into to be much clear to understand (Seperate lines and space breaks etc.)